### PR TITLE
Add appointment and the relevant interactions with patient

### LINKED ISSUE #97: [DEV] As a user, I want to view a specific doctor detail

### PR DESCRIPTION
**Pull request by ngsengleng**
### Summary

Resolves #97 

This PR implements the appointment feature as an extension of Activity, defined by an additional PatientId parameter and a unique clisyntax t/appointment. 

### Detail

Basically behaves the same as activity except for the extra patient id parameter. Is interlinked with patient, so you are not able to add an appointment if the patient id does not exist within the gomedic app and when a patient is deleted all relevant appointments with the same patient id will also be deleted as well. 
Graphically appointments are indicated by a visible patient id while activities are indicated by a lack of. this is the cleanest i can think of right now coz a placeholder patient id would mess up the clarity of the ui with a lot of irrelevant numbers.

### Pictures / Gif

Before adding
Screenshot 2021-10-20 at 5 55 33 PM

After adding
Screenshot 2021-10-20 at 5 56 54 PM

### Checks

- [x] Link PR to issue
- [x] if u dont mind, help me see if ive missed out anything
- [x] Add Reviewer
- [x] Pass CI

#### additional remarks
i'm gonna go and crash now soz ive been awake for 22 hours ill come back for the comments in the afternoon or evening. goodnight

### CODE REVIEW

File: src/main/java/gomedic/ui/card/ActivityCard.java [hunk 1]
```diff
@@ -43,6 +45,13 @@
     public ActivityCard(Activity activity) {
         super(FXML);
         this.activity = activity;
+        if (activity.isAppointment()) {
+            String pId = activity.getPatientId().toString();
+            //patientId.setText(String.format("Appointment with %s", pId));
```
Comment (simonjulianl) [inline src/main/java/gomedic/ui/card/ActivityCard.java:50]: You might want to delete this unused comments

File: src/main/java/gomedic/ui/card/ActivityCard.java [hunk 2]
```diff
@@ -43,6 +45,13 @@
...
+            patientId.setText("chickens");
```
Comment (simonjulianl) [inline src/main/java/gomedic/ui/card/ActivityCard.java:51]: Can we set this to a better text

File: src/main/java/gomedic/model/activity/Activity.java [hunk 3]
```diff
@@ -92,7 +129,14 @@ public boolean equals(Object other) {
 
     @Override
     public String toString() {
-        return activityId
+        return isAppointment
+                ? activityId
+                + "; Patient: " + patientId
+                + "; Title: " + title
```
[the remark this answers is not in the document]
Comment (ngsengleng) [reply src/main/java/gomedic/model/activity/Activity.java:135]: sorry i don't get quite understand this part. wdym by refactor isit i extract them out into two strings first? also do u mean to include isAppointment as part of the toString?

**Comment by ngsengleng** [comment]
sorry the conversations have been resolved

**Pull request merged**